packages/cms-data-sync/src/news/news.data-migration.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-console */
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { Entry, SysLink } from 'contentful-management';
+import { Document } from '@contentful/rich-text-types';
+
+import { newsQuery } from './news.queries';
+import {
+  Asset,
+  FetchNewsQuery,
+  FetchNewsQueryVariables,
+  Maybe,
+} from '../autogenerated-gql/graphql';
+
+import { migrateAsset } from '../assets/assets.data-migration';
+import {
+  convertText,
+  clearContentfulEntries,
+  getSquidexAndContentfulClients,
+  publishContentfulEntries,
+} from '../utils';
+
+type NewsPayload = {
+  title: { 'en-US': string };
+  shortText: { 'en-US': string | null };
+  frequency: { 'en-US': string | null };
+  link: { 'en-US': string | undefined };
+  linkText: { 'en-US': string | undefined };
+  text: { 'en-US': Document | null };
+  thumbnail: { 'en-US': SysLink | null };
+};
+
+const migrateNews = async () => {
+  const { contentfulEnvironment, squidexGraphqlClient } =
+    await getSquidexAndContentfulClients();
+
+  const convertThumbnail = async (
```

Comment:
i found it a little confusing that this function was declared inside `migrateNews`, though i understand it's because you want to be able to use stuff from it's scope (like `contentfulEnvironment`) - maybe it would be better to have it declared as factory and have it accept the `contentfulEnvironment`?

or just have the function accept it as the first argument, same as `clearContentfulEntries`

Reply:
It makes sense, I've committed in here https://github.com/yldio/asap-hub/pull/2290/commits/8abdb5e1a862cb415e625a97e28a70bc94851973